addon/MetadataStore.js

```diff
@@ -492,6 +492,22 @@ MetadataStore.prototype = {
   }),
 
   /**
+   * Check if the link exists in the database by checking if it's cache key exists
+   *
+   * @param {String} key a cache key
+   *
+   * Returns a promise with the array of the retrieved metadata record
+   */
+  asyncCacheKeyExists: Task.async(function*(key) {
```

Comment:
If this should function is meant to check that a record exists, it should only return a boolean.  Right now it looks like it returns an integer or an empty array, we should clear that up.